### DISCUSSION TITLE
Database Regions

### DISCUSSION THREAD

**Discussion by stexxart** [planetscale/discussion#66] [Feature Requests] [answered]
I have a few questions about where a database is deployed and the global distribution of the database infrastructure:

1. What region(s) are databases deployed to during the beta?
2. What region(s) will be supported in the near future, and is there a timeframe on this?
3. Are multi-region setups available?

We're looking to deploy a database in Australia & Ireland.

**Comment by iheanyi** [comment, marked as answer]
Hey there @stexxart, thanks for asking the question!

1. During the beta, databases are deployed to the eastern United States, in Northern Virignia.
2. Right now, we plan on adding support for Oregon and Dublin next, while we're still considering options for what further regions to add.
3. By multi-region, could you expand further? Your main database branch will be able to run in one region, but users will be able to create other branches in other regions if they so choose.

Let me know if you have any further questions!

**Reply by stexxart** [reply]
Hi @iheanyi, thanks for the response!

2. Any chance of Australia, PlanetScale could meet all our needs knowing that Ireland is now available?
3. That's clear, thanks. What about region-based sharding, so we can segment data by location?

**Reply by iheanyi** [reply]
@stexxart No problem at all! Sorry for the delayed response.

2) If there is enough demand to warrant supporting Australia, we maybe would. Demand isn't there yet. There will be options for provisioning a dedicated tenant for your organization in the Australia region, if you would be interested.
3) As for region-based sharding, this is on the roadmap but we don't have an estimated timeline for this just yet.

**Reply by abriginets** [reply]
I would say for covering the whole globe you would need to additionally have the following countries in your regions list:
- Bahrain (Middle East)
- Capetown (South Africa)
- Hong-Kong and Tokyo (Asia)
- Sydney (Australia)
- Sao Paulo (South America)

Also would definitely recommend adding Russia region as soon as they make it available.

**Comment by ghazninattarshah** [comment]
When is it coming to Asia/India? with support for multi region?

**Reply by mscoutermarsh** [reply]
Mumbai, Singapore and Tokyo are now available!

**Comment by stewarthsoj** [comment]
If you can get Sydney up and running, I'm sold!

**Reply by tbarn** [reply]
Just in case you missed it, we released it on Friday!

**Comment by akaufmann** [comment]
I hope that eu-central-1 (Frankfurt) will be added to the list in the near future. 🙂

**Reply by tbarn** [reply]
Just in case you missed it, we released it on Friday!

**Comment by bigint** [comment]
All this regions would be super cool, cuz vercel has it

PlanetScale and Vercel is a great combo 🤗 

<img width="420" alt="Screenshot 2021-09-08 at 9 25 13 PM" src="https://user-images.githubusercontent.com/69431456/132543260-6c73529b-fd45-4462-abc7-61d0e5967d53.png">

**Comment by irvile** [comment]
hi @iheanyi, any news about come to São Paulo, Brazil? we need you _**planetscale**_.

**Reply by tbarn** [reply]
Just in case you missed it, we released it on Friday!

**Comment by econic** [comment]
I'm worrying that the lack of `Frankfurt, Germany` is actually a dealbreaker for us.
Cloudping reports ~30 for Frankfurt<>Dublin, so getting this down to 3 is quite the difference.

But we'll do some testing, maybe it works...

**Reply by tiaringhio** [reply]
Would also love a location in Germany, Dublin works but Frankfurt would be way better for southern European infrastructure

**Reply by tbarn** [reply]
Frankfurt region now exists! We just released it on Friday.

**Comment by FabrizioTorrico** [comment]
Don't forget south america!🇦🇷

**Reply by tbarn** [reply]
Not exactly where you are requesting, but as of Friday we have a São Paulo region.

**Comment by fooforge** [comment]
Hey all,

I'm happy to announce that we have [added three more regions](https://planetscale.com/changelog?featured=de-au-br-regions) across the globe: Frankfurt (eu-central-1), Sydney (ap-southeast-2), and Sao Paulo (sa-east-1). Check these out, and if you hit any snags [reach out](https://planetscale.com/support). Thanks! 😄

**Reply by stewarthsoj** [reply]
Amazing @fooforge! 🔥

**Comment by lewis45454** [comment]
When will there be a London region?

**Reply by tbarn** [reply]
Just in case you missed it, AWS eu-west-2 (London) was added a few months ago.

**Comment by e2goon** [comment]
Seoul region? 😢

**Comment by devhyunjae** [comment]
We neeeeeed Seoul 🤩

**Comment by axyjo** [comment]
Any chance of deploying to ca-central-1?

**Reply by brettr91** [reply]
Yeah, this would be great. Really can't use the service within Canada, as most private organizations require data to be held within Canada.

**Reply by jmarus** [reply]
+1 for Canada. They really should add at least once Canadian region.

**Reply by sunappio** [reply]
+1 for Canada

**Comment by asagtc** [comment]
Hello, Any plans to add Bahrain?

**Comment by Suya1671** [comment]
af-south-1 would be incredibly helpful

**Comment by devunt** [comment]
Not having Seoul (ap-northeast-2) region is a dealbreaker to adapt Planetscale (we are using pgsql for now but we are now even considering migrating to mysql if Planetscale had Seoul region)

**Reply by tbarn** [reply]
We are looking into adding this region. No timeline, but it's definitely being considered.

**Reply by jeffminsungkim** [reply]
> We are looking into adding this region. No timeline, but it's definitely being considered.

Any updates on Seoul region support? Even a rough timeline would be helpful.

**Reply by tbarn** [reply]
It's in progress!

**Comment by RealPresidentlin** [comment]
What would it take to get South Africa on the list, we currently have AWS Cape Town (cpt1)

I just moved my vercel functions to SA and have seen much improved speeds but my PS DB is in London, previously both were using London.

I'll probably move back to London after testing.

**Reply by KrisStadler** [reply]
@Lwachira interested to know the performance differences between having the serverless functions in London vs Cape Town with the DB based in London. Would be nice if you can share some details of your finding here please. :smile:

**Reply by RealPresidentlin** [reply]
Hi! @KrisStadler , no problem, the app is way faster at initial load when it's in CPT and the planetscale db is in London, each DB query takes seconds instead of milliseconds, levels of 4 - 10 seconds, that's mostly cause of the distance. 

When they are next to each other, the app is way faster (more so now thanks to the work the Prisma team has done), cold starts are brutal for my app but after that, it's fairly quick, milliseconds instead of seconds. I've perfected the queries to return 100% filtered data, so that helped as well.

A golden win would be getting a db instance in South Africa, in a year ps or someone else will figure this out, either ps will figure out edge db or someone else will.

**Reply by KrisStadler** [reply]
Thanks for the feedback @Lwachira, also noticed the 4 - 10 second delay of DB queries between CPT and London. Probably going to replicate something similar while I wait for the ps CPT region to come online.

**Comment by KrisStadler** [comment]
Also want to know the status on AWS Cape Town region. My Vercel application is configured to use CPT serverless functions, but the closest I can get the DB is London :(

**Comment by paulm17** [comment]
Is there an official regions list?

**Reply by tbarn** [reply]
Yup! https://planetscale.com/docs/concepts/regions#available-regions

**Comment by pjlast** [comment]
Would like to add another voice to the "please open up the Cape Town region" list, as AWS does have an `af-south-1` region (as already mentioned) 🙏

**Comment by Reghardt** [comment]
Any news on a Cape Town Region? It will really make life easier for devs this side of the world.

**Comment by kimagroup** [comment]
Hello, is there possibility to add Bahrain and Cape town regions, which are really important for making everything global?

**Comment by ciacka** [comment]
Waiting for `europe-central2` (GCP based in Warsaw, Poland 🇵🇱) 🫣

**Comment by shaun-fitzgibbon** [comment]
Any News on adding a af-south-1 location ?

**Reply by RealPresidentlin** [reply]
Thank you for prompting again. My gut instinct is there is many of us but it's hard to see everyone to justify the business expansion. The more people who add their 2cents the better.

**Comment by ryush00** [comment]
We are waiting for GCP tokyo region for PostgreSQL.
